src/usecases/webhook-use-case.ts

```diff
@@ -0,0 +1,152 @@
+import type {
+	AssociatedFigmaDesign,
+	ConnectInstallation,
+	FigmaTeam,
+} from '../domain/entities';
+import { WebhookEventType } from '../domain/entities';
+import { figmaService } from '../infrastructure/figma';
+import { jiraService } from '../infrastructure/jira';
+import {
+	associatedFigmaDesignRepository,
+	connectInstallationRepository,
+	figmaTeamRepository,
+	RepositoryRecordNotFoundError,
+} from '../infrastructure/repositories';
+
+const SUPPORTED_WEBHOOK_EVENT_TYPES: Set<WebhookEventType> = new Set([
+	WebhookEventType.FILE_UPDATE,
+	WebhookEventType.FILE_DELETE,
+]);
+
+export const webhookUseCase = {
+	execute: async (
+		webhookId: string,
+		fileKey: string,
+		eventType: WebhookEventType,
+	): Promise<void> => {
+		if (!SUPPORTED_WEBHOOK_EVENT_TYPES.has(eventType)) {
+			return;
+		}
+
+		const figmaTeam = await figmaTeamRepository.getByWebhookId(webhookId);
+		const [connectInstallation, associatedFigmaDesigns] = await Promise.all([
+			connectInstallationRepository.get(figmaTeam.connectInstallationId),
+			associatedFigmaDesignRepository.findManyByFileKeyAndConnectInstallationId(
+				fileKey,
+				figmaTeam.connectInstallationId,
+			),
+		]);
+
+		// If an error is thrown while handling a webhook, Figma will retry
+		// sending the event up to 3 times at 5/30/180 minutes. Handlers should
+		// be idempotent to handle this gracefully.
+		if (eventType === WebhookEventType.FILE_UPDATE) {
+			return handleFileUpdateEvent(
+				associatedFigmaDesigns,
+				figmaTeam,
+				connectInstallation,
+			);
+		}
+
+		if (eventType === WebhookEventType.FILE_DELETE) {
+			return handleFileDeleteEvent(associatedFigmaDesigns, connectInstallation);
+		}
+	},
+};
+
+const handleFileUpdateEvent = async (
+	associatedFigmaDesigns: AssociatedFigmaDesign[],
+	figmaTeam: FigmaTeam,
+	connectInstallation: ConnectInstallation,
+): Promise<void> => {
+	const designs = await Promise.all(
+		associatedFigmaDesigns.map((design) =>
+			figmaService.fetchDesignById(
+				design.designId,
+				figmaTeam.figmaAdminAtlassianUserId,
+			),
+		),
+	);
+
+	await jiraService.updateDesigns(designs, connectInstallation);
+};
+
+const handleFileDeleteEvent = async (
+	associatedFigmaDesigns: AssociatedFigmaDesign[],
+	connectInstallation: ConnectInstallation,
+): Promise<void> => {
+	// Delete designs from Jira
+	const { rejected, fulfilled } = await settlePromisesAndSortResults(
```

Comment:
We may also need to think about the phone home (creating or deleting dev resources) and writing to issue properties as part of these webhook flows